edge-util/src/Microsoft.Azure.Devices.Edge.Storage.RocksDb/disk/DiskSpaceChecker.cs

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Storage.RocksDb.Disk
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Storage.Disk;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class DiskSpaceChecker : IStorageSpaceChecker
+    {
+        const int CheckFrequencyDefault = 120;
+        readonly string storageFolder;
+        readonly object updateLock = new object();
+        readonly int checkFrequency;
+        Option<DiskSpaceCheckerBase> inner;
+
+        DiskSpaceChecker(string storageFolder, Option<DiskSpaceCheckerBase> inner, int checkFrequency)
+        {
+            this.checkFrequency = checkFrequency;
+            this.storageFolder = Preconditions.CheckNonWhiteSpace(storageFolder, nameof(storageFolder));
+            this.inner = Preconditions.CheckNotNull(inner, nameof(inner));
+        }
+
+        // Returns false if checker is disabled (i.e. Option.HasValue is false)
+        public bool IsFull => this.inner.Match<bool>(b => b.DiskStatus == DiskSpaceStatus.Full, () => false);
+
+        public static DiskSpaceChecker Create(string storageFolder, Option<int> checkFrequency)
+        {
+            // Start up diskSpaceChecker unfilled - so that it will be enabled when SetMaxSizeBytes is called
+            var diskSpaceChecker = new DiskSpaceChecker(storageFolder, Option.None<DiskSpaceCheckerBase>(), checkFrequency.GetOrElse(CheckFrequencyDefault));
+            return diskSpaceChecker;
+        }
+
+        public void SetMaxSizeBytes(Option<long> maxSizeBytes)
+        {
+            if (maxSizeBytes.HasValue)
```

Comment:
I think you should be able to use-

this.inner = maxSizeBytes.Match(...);

where 'some' creates a new Option and returns it.
'none' returns Option.None.

You will need to dispose the original PeriodicTask for either case so place that before the match maybe. Also, you do need locking to account for parallel calls to the SetMaxSizeBytes method.